packages/bls-kms-local/src/agent/BlsKeyManager.ts

```diff
@@ -0,0 +1,40 @@
+import { KeyManager, AbstractKeyManagementSystem, AbstractKeyStore } from '@veramo/key-manager'
+
+import {
+    IKey, IKeyManagerSignArgs, TKeyType
+} from '@veramo/core'
+
+export class BlsKeyManager extends KeyManager {
+
+    private localStore: AbstractKeyStore
+    private localKms: Record<string, AbstractKeyManagementSystem>
+
+    constructor(options: { store: AbstractKeyStore; kms: Record<string, AbstractKeyManagementSystem> }) {
+        super({ store: options.store, kms: options.kms})
+        this.localStore = options.store
+        this.localKms = options.kms
+    }
+
+    private getLocalKms(name: string): AbstractKeyManagementSystem {
+        const kms = this.localKms[name]
+        if (!kms) {
+            throw Error(`invalid_argument: This agent has no registered KeyManagementSystem with name='${name}'`)
+        }
+        return kms
+    }
+
+    async keyManagerSign(args: Omit<IKeyManagerSignArgs, 'data'> & { data: string | string[]}): Promise<string> {
+        const keyInfo: IKey = await this.localStore.get({ kid: args.keyRef }) as IKey
+        const kms = this.getLocalKms(keyInfo.kms)
+        let input: any;
+        if (Array.isArray(args.data)) {
```

Comment:
ternary would be nicer

Reply:
Fixed!

Reply:
haven't looked at the changes yet. But do not use any. If there are multiple types possible, state those multiple types. if you really have no clue what it could be, use unknown

Reply:
Just a dirty hack to be able to insert an array without changing the original interface. Removed the array, so I think it will work fine